Feature/home page

PatrykKeska opened the pull request.

PatrykKeska committed:
* feature/HomePage branch created

PatrykKeska committed:
* Branch feature/HomePage created
* React Router has been implemented
* Basic atomic file layout added

PatrykKeska committed:
* Responsive font-size added to global style provider

PatrykKeska committed:
* HomePage Component created which implements HomePageTemplate

PatrykKeska committed:
* Paragraph component created

PatrykKeska committed:
* HomePage Logo component created

PatrykKeska committed:
* HomePage Logo asset added

PatrykKeska committed:
* Link component created( <a>Link</a>)
* basic prop types implemented

PatrykKeska committed:
* Button component created

PatrykKeska committed:
* styles hotfix
* cleanup props

PatrykKeska committed:
* Input component created

PatrykKeska committed:
* Wrapper component created

PatrykKeska committed:
* LoginForm Created ( Molecules)
Login Form include :
- Inputs
- Links
- Button

PatrykKeska committed:
* HomePage template created

PatrykKeska committed:
* HomePage Organism created

joanbier reviewed (commented):
Jak dla mnie wygląda wszystko ok, chociaż nie ogarniam tej biblioteki Atom, niech się inni wypowiedzą ;D

joanbier commented on src/Atoms/Input/Input.tsx:
```diff
@@ -0,0 +1,15 @@
+
+
+
+export const Input = styled.input`
```
możemy tutaj zmienić width na jednostkę względną np. vw. Będzie wtedy bardziej RWD i ładnie się skalowało na netbooka czy większego tableta, lub odpowienio duże na większych monitorach typu 28cal. W AdobeXD mockup mamy ustawione na 1920 × 1080, łatwo obliczyć proporcję. W typ przypadku width możemy zapisać jako 18vw.

joanbier commented on src/Atoms/LogoMegaK/LogoMegaK.tsx:
```diff
@@ -0,0 +1,13 @@
+  background-size: cover;
+  background-repeat: no-repeat;
+  margin-bottom: 30px;
+  width: 124px;
```
tutaj można zastosować podobnie jak z inputem, wpisać w  % lub vw

joanbier commented on src/Atoms/LogoMegaK/LogoMegaK.tsx:
```diff
@@ -0,0 +1,13 @@
+  background-repeat: no-repeat;
+  margin-bottom: 30px;
+  width: 124px;
+  height: 76px;
```
heigh wtedy na auto, bedzie się automatycznie saklować

joanbier commented on src/Atoms/Wrapper/Wrapper.tsx:
```diff
@@ -0,0 +1,14 @@
+import styled from "styled-components";
+
+export const Wrapper = styled.div`
```
Trzeba się pozbyć scrollbarów pionowych i poziomych. Zamiast min-height zastosować max-height lub overflow hidden

joanbier commented:
Co do mobile, dobrze zrobić media na 768px i dać np inputy na 90vw etc.

joanbier commented:
Dopatrzyłam się jeszcze, że brakuje font-family 'Catarman' oraz odpowiedni font-size jak w mockupie

PatrykKeska committed:
* hotfix Input, Logo, => width and height use scaling unit instead of pixels

* GlobalStyle => body use overflow:hidden to remove scroll bars

PatrykKeska committed:
* hotfix Input, Logo, => width and height use scaling unit instead of pixels

* GlobalStyle => body use overflow:hidden to remove scroll bars

*index.css and app.css removed since we have GlobalStyles.ts

*Font Family changed to Caterman

* Font saved to assets/fonts

zannien94 commented on src/Atoms/Input/Input.tsx:
```diff
@@ -0,0 +1,16 @@
+import styled from "styled-components";
```
jezeli robisz plik Input to po co folder jezeli nic wiecej niema ? 
jezeli juz tworzymy folder dla Input to fajnie by bylo zrobic plik do exportu np. index.js wtedy unikamy importu Input/Input -https://github.com/airbnb/javascript

PatrykKeska replied:
Wprowadziłem te zmiany o których pisałaś, zostawiłem folder przy formularzu logowania z racji że na pewno będą jakieś funkcje dodane

zannien94 commented on src/Atoms/LoginButton/LoginButton.tsx:
```diff
@@ -0,0 +1,20 @@
+import React from "react";
```
to samo co powyzej

zannien94 commented on src/Atoms/LoginButton/LoginButton.tsx:
```diff
@@ -0,0 +1,20 @@
+export const LoginButton = () => {
+
+    return (
+        <Button>Zaloguj Się</Button>
```
jak myslisz Button bedzie chyba uzywany w nie jednym miejscu ? 
moze warto by bylo zrobic kompontent Button i przekazac do niego propsa ? text ?

zannien94 commented on src/Atoms/LogoMegaK/LogoMegaK.tsx:
```diff
@@ -0,0 +1,15 @@
+import styled from "styled-components";
```
to samo co powyzej

zannien94 commented on src/Atoms/Paragraph/Paragraph.tsx:
```diff
@@ -0,0 +1,6 @@
+import styled from "styled-components";
```
to samo co powyzej

zannien94 commented on src/Atoms/StyledLink/Link.tsx:
```diff
@@ -0,0 +1,30 @@
+import styled from "styled-components";
```
to samo co powyzej

zannien94 commented on src/Atoms/StyledLink/Link.tsx:
```diff
@@ -0,0 +1,30 @@
+}
+
+export const StylesLink = styled(Link)<Props>`
+
```
jezeli uzywamy 2x props moze warto zrobic jeden wrapper ? 
${({ align, underline }) => css`
  color: white;
  align-self:${align} ;
  text-decoration: ${underline || "none"};
`}

zannien94 commented on src/Atoms/StyledLink/Link.tsx:
```diff
@@ -0,0 +1,30 @@
+  
+`
+
+export const StyledLink = (props:Props)=>{
```
fajnie by bylo zdestrukturyzowac propsy

zannien94 commented on src/Atoms/Wrapper/Wrapper.tsx:
```diff
@@ -0,0 +1,14 @@
+import styled from "styled-components";
```
to samo co powyzej , ogolnie zerknij na wiekszosc plikow

zannien94 commented on src/Molecules/LoginForm/LoginForm.tsx:
```diff
@@ -0,0 +1,64 @@
+import styled from "styled-components";
+import {Input} from "../../Atoms/Input/Input";
+import {StyledLink} from "../../Atoms/StyledLink/Link";
+import React from "react";
```
importy absolutne powinny raczej byc przy importach absolutnych 

import absolute
import relative 

czyli pierw importy z node modules a potem swoje komponenty

zannien94 commented on src/Molecules/LoginForm/LoginForm.tsx:
```diff
@@ -0,0 +1,64 @@
+
+        <StyledForm>
+            <Input
+                type={"email"}
```
wywal { - brackety nic tutaj nie wnosza

zannien94 commented on src/index.tsx:
```diff
@@ -1,18 +1,19 @@
 import reportWebVitals from './reportWebVitals';
-import {ThemeProvider} from "styled-components";
 import {GlobalStyle} from "./utils/theme/GlobalStyles";
+import {BrowserRouter} from "react-router-dom";
```
prosze wynies na poczatek pilku

PatrykKeska committed:
* code review hotfix

PatrykKeska committed:
* hover effect added with cursor pointer on button

PatrykKeska committed:
* File structure changed :

Components Folder include :

* Atoms
* Molecules
* Templates

Basic structure is :

Atoms => Molecules => Templates => Pages

This is how we're going to scale app

PatrykKeska committed:
* File structure changed :

Components Folder include :

* Atoms
* Molecules
* Organism

Basic structure is :

Atoms => Molecules => Organism => Pages

This is how we're going to scale app

PatrykKeska commented:
Jężeli nie ma wiekszych/mniejszych problemów to leci marge do developa, z racji że już powstałe komponenty będą re używalne w następnych widokach
